pkg/destroy/aws/aws.go

```diff
@@ -165,132 +162,322 @@ func (o *ClusterUninstaller) RunWithContext(ctx context.Context) error {
 		logger:  o.Logger,
 	}
 
-	deleted, err := terminateEC2InstancesByTags(ctx, ec2.New(awsSession), iamClient, o.Filters, o.Logger)
+	// Get the initial resources to delete, so that they can be returned if the context is canceled while terminating
+	// instances.
+	deleted := sets.NewString()
+	resourcesToDelete, tagClientsWithResources, err := o.findResourcesToDelete(ctx, tagClients, iamClient, iamRoleSearch, iamUserSearch, deleted)
 	if err != nil {
-		return err
+		o.Logger.WithError(err).Info("error while finding resources to delete")
+		if err := ctx.Err(); err != nil {
+			return resourcesToDelete.UnsortedList(), err
+		}
 	}
 
 	tracker := new(errorTracker)
-	tagClientStack := append([]*resourcegroupstaggingapi.ResourceGroupsTaggingAPI(nil), tagClients...)
+
+	// Terminate EC2 instances.
+	ec2Client := ec2.New(awsSession)
```

Comment:
I think this needs a region argument otherwise this might not function correctly..?

`aws.NewConfig().WithRegion(region)` something like that..

Reply:
The current code in master does not specify the region. The `awsSession` should already be configured with the region.
https://github.com/openshift/installer/blob/d91c40bca7952bd89730991f949cb10d16a71f9b/pkg/destroy/aws/aws.go#L168